test/integration/rekor_monitor_integration_test.go

```diff
@@ -91,6 +92,70 @@ func TestTamperedCheckpoint(t *testing.T) {
 	stopMonitor(t, runCmd)
 }
 
+func TestLogTruncationForking(t *testing.T) {
+	ctx, binary, checkpointFile, monitorPort, mockServer := setupTest(t)
+	defer mockServer.Close()
+
+	t.Run("truncate_fork_checkpoint_file", func(t *testing.T) {
+		content, err := os.ReadFile(checkpointFile)
+		if err != nil {
+			t.Fatalf("failed to read checkpoint file: %v", err)
+		}
+
+		contentStr := string(content)
```

Comment:
The string manipulation is very complex and hard to read. Please extract it to a helper function with proper documentation on what it does and why you use the "magic numbers"